lib/zwave/system/capabilities/dim/BASIC.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+module.exports = {
+	get: 'BASIC_GET',
+	set: 'BASIC_SET',
+	setParser(value) {
+		if (this.hasCapability('onoff')) this.setCapabilityValue('onoff', value > 0);
+		return {
+			Value: Math.round(value * 99),
+		};
+	}),
+	report: 'BASIC_REPORT',
+	reportParser: report => {
```

Comment:
I don't think this works, did you test it? The scope in the `reportParser` is not correct, `this` is not the device instance. You need to format your function similar to the `setParser`.

Reply:
mmm i did not test when added in the system capabilities.
does work in my testing code for one of the devices i've worked on a little though still without setting onoff:
https://github.com/caseda/com.hank-tech/blob/temp_color_switch/drivers/HKZW-RGB01/device.js#L31
so you might be right that `this` is not going to be in the scope now